scipy/stats/_continuous_distns.py

```diff
@@ -5478,10 +5478,27 @@ def _entropy(self):
         return 0.5 * np.log(2 * np.pi) + 0.5
 
 
+# gilbrat was a spelling error; correct is gibrat, see #15911
 gilbrat = gibrat_gen(a=0.0, name='gilbrat')
 gibrat = gibrat_gen(a=0.0, name='gibrat')
 
 
+# since the deprecated class gets intantiated upon import (and we only want to
+# warn upon use), add the deprecation to each (documented) class method, c.f.
+# https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gilbrat.html
+_gibrat_method_names = [
+    "cdf", "entropy", "expect", "fit", "interval", "isf", "logcdf", "logpdf",
+    "logsf", "mean", "median", "moment", "pdf", "ppf", "rvs", "sf", "stats",
+    "std", "var"
+]
```

Comment:
@h-vetinari whenever you have a chance - I don't think this raises a deprecation warning when `gilbrat` is frozen or when any of the frozen distribution methods get called. Let's have this raise a deprecation warning when `gilbrat` gets frozen (but I don't think we need to warn on every method call of the frozen distribution).

Reply:
Hey @mdhaber 

I hadn't looked that deep into the infrastructure, but looking at [`rv_continuous.__init__`](https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L1895), I don't see where the methods would be overwritten?

There _is_ a call to [`self._attach_methods()`](https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L1931), but that doesn't [seem](https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L1961) to overwrite the methods we're deprecating here.

Have you tried that this doesn't work on main?

Reply:
Yup, I tried. We need to deprecate the `__call__` method. I found this when I started using the same approach in gh-16042 (line 1582). Adding `__call__` to the list of methods didn't seem to work so I added the decorator to the method directly.

Reply:
```python3
from scipy.stats import gilbrat
import warnings
warnings.simplefilter('default')
gilbrat() # <scipy.stats._distn_infrastructure.rv_continuous_frozen at 0x1ae11002700>
gilbrat.pdf(1)  #  0.3989422804014327
C:\Users\matth\AppData\Local\Temp\ipykernel_24796\4065156008.py:1: DeprecationWarning:         `gilbrat.pdf` is deprecated, use `gibrat.pdf` instead!
        Please replace all uses of the distribution class `gilbrat` with the corrected spelling `gibrat`. `gilbrat` will be removed in SciPy 1.11.
  gilbrat.pdf(1)
```

Reply:
@mdhaber 
I'm sorry this slipped under the radar. I'm trying to fix this in #16257. 

> Adding `__call__` to the list of methods didn't seem to work, so I added the `np.deprecate` decorator to the method directly.

I'd be happy to know how you did that (as I don't see a fundamental difference between doing it directly and adding "__call__" to the list).